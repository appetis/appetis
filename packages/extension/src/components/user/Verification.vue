<template>
  <div
    class="fixed inset-0 overflow-y-auto z-10"
    aria-labelledby="modal-title"
    role="dialog"
    aria-modal="true"
  >
    <div class="init-modal-background" aria-hidden="true"></div>
    <div class="modal">
      <!--      <span class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true">&#8203;</span>-->

      <div class="modal-content-sm">
        <div class="modal-body">
          <h2 class="text-lg py-4">Verification</h2>
          <p>
            Verification code has been sent to your email address.<br />
            Please check your email.
          </p>

          <div class="flex">
            <!--            <CodeInput @codeKeyup="codeKeyup" :current="1" />-->
            <div class="w-1/6 px-1">
              <input
                type="text"
                maxlength="1"
                class="input-underline"
                ref="code_1"
                @keyup="codeKeyup(1)"
              />
            </div>
            <div class="w-1/6 px-1">
              <input
                type="text"
                maxlength="1"
                class="input-underline"
                ref="code_2"
                @keyup="codeKeyup(2)"
              />
            </div>
            <div class="w-1/6 px-1">
              <input
                type="text"
                maxlength="1"
                class="input-underline"
                ref="code_3"
                @keyup="codeKeyup(3)"
              />
            </div>
            <div class="w-1/6 px-1">
              <input
                type="text"
                maxlength="1"
                class="input-underline"
                ref="code_4"
                @keyup="codeKeyup(4)"
              />
            </div>
            <div class="w-1/6 px-1">
              <input
                type="text"
                maxlength="1"
                class="input-underline"
                ref="code_5"
                @keyup="codeKeyup(5)"
              />
            </div>
            <div class="w-1/6 px-1">
              <input
                type="text"
                maxlength="1"
                class="input-underline"
                ref="code_6"
                @keyup="codeKeyup(6)"
              />
            </div>
          </div>

          <div class="text-right flex mt-6">
            <div class="w-full px-1">
              <button class="btn-black" @click="verifyCode">Continue</button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
//import CodeInput from "./CodeInput";
export default {
  name: 'Verification',
  components: {
    //  CodeInput
  },
  props: ['callFocus'],
  data() {
    return {
      current: 1,
    }
  },
  mounted() {
    // if(this.showCode) this.focusInput()
  },
  watch: {
    callFocus() {
      this.focusInput()
    },
  },
  methods: {
    focusInput() {
      //this.$refs['code_1'].focus();
      this.goNextFocus('code_1')
    },
    goNextFocus(next) {
      this.$refs[next].focus()
    },
    codeKeyup(num) {
      this.current = num
      if (this.$refs['code_' + num].value) {
        if (this.current < 6) this.current++
      } else {
        if (this.current > 1) this.current--
      }
      this.goNextFocus('code_' + this.current)
    },
    verifyCode() {
      console.log('verifyCode')
    },
    goSignup() {
      this.$emit('goSignup')
    },
  },
}
</script>
