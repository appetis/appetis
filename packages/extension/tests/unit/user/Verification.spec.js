import Verification from '@/components/user/Verification'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

let store

beforeEach(() => {
  jest.clearAllMocks()

  store = new Vuex.Store({
    state: {
      user: {
        id: 2,
        teams: [{ id: 1 }],
      },
      isGuest: false,
    },
  })
})

const factory = () => {
  return mount(Verification, { store, localVue })
}

describe('init/Verification', () => {
  it('verification modal snapshot', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('click Verification button', async () => {
    const wrapper = factory()
    const mockOnSubmit = jest.spyOn(wrapper.vm, 'onSubmit')
    await wrapper.find('form').trigger('submit')
    //await wrapper.find('#btn-code-continue').trigger('click')

    expect(mockOnSubmit).toHaveBeenCalled()
  })
})
