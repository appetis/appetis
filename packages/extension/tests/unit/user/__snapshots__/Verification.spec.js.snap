// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init/Verification verification modal snapshot 1`] = `
<div aria-labelledby="modal-title" role="dialog" aria-modal="true" class="fixed inset-0 overflow-y-auto z-10">
  <div aria-hidden="true" class="init-modal-background"></div>
  <div class="modal">
    <div class="modal-content-sm">
      <div class="modal-header init-modal-header">
        <div class="w-1/10 bg-wk-gray-box px-2 rounded mr-2 align-middle pt-1">
          W
        </div>
        <div class="w-9/10 pt-1">Unlock Workly with Verification Code</div>
      </div>
      <div class="modal-body">
        <p class="text-sm">
          Verification code has been sent to your email address. Please check
          your email.
        </p>
        <form>
          <div class="flex mt-6">
            <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
            <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
            <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
            <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
            <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
            <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
          </div>
          <div class="
              text-red-700
              mt-4
              border border-red-700
              rounded
              w-full
              py-2
              px-3
            " style="display: none;">

          </div>
          <div class="mt-7"><button class="btn-black-full">
              Continue
            </button>
            <div class="spinner spinner--circle-2 mx-auto" style="width: 40px; height: 40px; border: 5px solid #41b883; border-top-color: #35495e; display: none;"></div>
          </div>
        </form>
      </div>
      <div class="modal-footer init-modal-footer"></div>
    </div>
  </div>
</div>
`;
